examples/Example/Page/Todos/TodoCSS.hs

```diff
@@ -0,0 +1,274 @@
+                . style' "color: #b83f45"
+            )
+            "index"
+
+--- TodosView ----------------------------------------------------------------------------
+
+data AllTodos = AllTodos
+  deriving (Generic, ViewId)
+
+instance (Todos :> es) => HyperView AllTodos es where
+  type Require AllTodos = '[TodoView]
+
+  data Action AllTodos
+    = ClearCompleted
+    | Filter FilterTodo
+    | SubmitTodo
+    | ToggleAll FilterTodo
+    | SetCompleted FilterTodo Todo Bool
+    deriving (Generic, ViewAction)
+
+  update = \case
+    SubmitTodo -> do
+      TodoForm task <- formData @(TodoForm Identity)
+      _ <- Todos.create task
+      todos <- Todos.loadAll
+      pure $ todosView FilterAll todos
+    ToggleAll filt -> do
+      todos <- filteredTodos filt
+      updated <- Todos.toggleAll todos
+      pure $ todosView filt updated
+    ClearCompleted -> do
+      todos <- Todos.clearCompleted
+      pure $ todosView FilterAll todos
+    Filter filt -> do
+      todos <- filteredTodos filt
+      pure $ todosView filt todos
+    SetCompleted filt todo completed -> do
+      _ <- Todos.setCompleted completed todo
+      todos <- filteredTodos filt
+      pure $ todosView filt todos
+   where
+    filteredTodos filt =
+      filter (isFilter filt) <$> Todos.loadAll
+
+    isFilter filt todo =
+      case filt of
+        FilterAll -> True
+        Active -> not todo.completed
+        Completed -> todo.completed
+
+todosView :: FilterTodo -> [Todo] -> View AllTodos ()
+todosView filt todos = do
+  header (extClass "header") $ do
+    h1 (style' "top:-80px") $ text "todos"
+    todoForm
+  main' (extClass "main") $ do
+    div' (extClass "toggle-all-container") $ do
+      input'
+        ( extClass "toggle-all"
+            . att "id" "toggle-all"
+            . att "type" "checkbox"
+        )
+      label'
+        ( extClass "toggle-all-label"
+            . att "for" "toggle-all"
+            . onClick (ToggleAll filt)
+        )
+        (text "Mark all as complete")
+    ul' (extClass "todo-list") $ do
+      forM_ todos $ \todo -> do
+        hyper (TodoView todo.id) $ todoView filt todo
+  statusBar filt todos
+
+todoForm :: View AllTodos ()
+todoForm = do
+  let f :: TodoForm FieldName = fieldNames
+  form SubmitTodo grow $ do
+    field f.task id $ do
+      Input (FieldName nm) <- context
+      input' -- we use a custom input field, because the Hyperbole one overrides autocomplete
+        ( extClass "new-todo"
+            {-
+              -- . autofocus
+              FIXME: turning off autofocus, that "steals" the focus on item click.
+              FIXME: to solve this, we could either store the "initially focused" state and track that boolean, or use buttons
+              FIXME: but since this example is meant to match as close as possible to the original CSS version
+              FIXME: and not diverge too much from the other todo example, I'm leaving as-is.
+             -}
+            . att "autocomplete" "off"
```

Comment:
We should go fix the source for `input`, right? Probably set to "off" most of the time. (Right now `input TextInput`, which is in the main example, sets it to "text-input" which is wrong). 

I wanted to make it easy to set "type" and "autocomplete" with one type, but maybe it wasn't a great choice?